public/app/features/provisioning/hooks/useCreateOrUpdateRepository.ts

```diff
@@ -1,37 +1,78 @@
 import { useCallback } from 'react';
 
+import { AppEvents } from '@grafana/data';
+import { getAppEvents } from '@grafana/runtime';
 import {
   RepositorySpec,
   useCreateRepositoryMutation,
+  useCreateRepositoryTestMutation,
   useReplaceRepositoryMutation,
 } from 'app/api/clients/provisioning';
 
 export function useCreateOrUpdateRepository(name?: string) {
   const [create, createRequest] = useCreateRepositoryMutation();
   const [update, updateRequest] = useReplaceRepositoryMutation();
+  const [testConfig, testRequest] = useCreateRepositoryTestMutation();
+  const appEvents = getAppEvents();
 
   const updateOrCreate = useCallback(
-    (data: RepositorySpec) => {
-      if (name) {
-        return update({
-          name,
-          repository: {
-            metadata: {
-              name,
-              // TODO? -- replace with patch spec, so the rest of the metadata is not replaced?
-              // Can that support optimistic locking? (eg, make sure the RV is the same?)
-              finalizers: ['cleanup', 'remove-orphan-resources'],
-            },
+    async (data: RepositorySpec) => {
+      try {
+        // First test the config
+        const testResult = await testConfig({
+          // HACK: we need to provide a name to the test configuration
+          name: name || 'new',
+          body: {
             spec: data,
           },
+        }).unwrap();
+
+        if (!testResult.success) {
+          appEvents.publish({
+            type: AppEvents.alertError.name,
+            payload: ['Repository configuration test failed', testResult.errors?.join(', ')],
+          });
+          return;
```

Comment:
I'm a bit hesitant to emit these events here, I think it'd be done in places where this hook is consumed. The same goes for the error handling.

Reply:
let's then hold this PR till this one is merged: https://github.com/grafana/grafana/pull/103290/files#diff-47b52331a48eb9157cf74fade15880d10a9cab802b92bb20c528ec3b28eabcb3